native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -118,30 +119,68 @@ public static Object convertToJSON(BXml xml, String attributePrefix, boolean pre
      * @return ObjectNode Json object node corresponding to the given xml element
      */
     private static Object convertElement(BXmlItem xmlItem, String attributePrefix,
-                                         boolean preserveNamespaces) {
+                                         boolean preserveNamespaces, AttributeManager attributeManager) {
         BMap<BString, Object> rootNode = newJsonMap();
-        LinkedHashMap<String, String> attributeMap = collectAttributesAndNamespaces(xmlItem, preserveNamespaces);
+        BMap<BString, Object> mapData = newJsonMap();
+        if (preserveNamespaces) {
+            getAttributes(xmlItem, preserveNamespaces, attributePrefix, mapData, attributeManager);
```

Comment:
```suggestion
            processAttributes(xmlItem, preserveNamespaces, attributePrefix, mapData, attributeManager);
```
Since this function is not returning any values.